### DISCUSSION TITLE
Using mesh generated by gmsh

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#23396] [Q&A Meshing] [answered]
I also need a mesh as in [discussion](https://github.com/idaholab/moose/discussions/16494) and used the same .geo file to generate a mesh. But I have trouble defining BCs by using this mesh.  It always reports an error "boundary top can't be found". I checked mesh using peacock and found top, bottom and so on are blocks not boundaries. So how should I define these as boundaries in gmsh? What is the correspondence between gmsh objects and MOOSE objects? This is really not documented in detail.

**Comment by RWTHLHK** [comment, marked as answer]
The solution is to add a phsical surface, then this physical surface will be in block. The other physical curves are in boundaries.

**Comment by Rahim-Habibi** [comment]
Hi, 
can you share your final .geo file here.
